app/models/driver.rb

```diff
@@ -0,0 +1,52 @@
+class Driver < ApplicationRecord
+  has_many :trips
+
+  validates :name, presence: true
+  validates :vin, presence: true, length: { is: 17 }
+  validates :available, inclusion: { in: ["true", "false"] }
+
+  def average_rating
+    nil_counter = 0
+    rating_sum = 0
+    self.trips.each do |trip|
+      if trip.rating.nil?
+        nil_counter += 1
+      else
+        rating_sum += trip.rating.to_f
+      end
+    end
+
+    num_ratings = (self.trips.length) - nil_counter
+    average = (rating_sum/num_ratings.to_f).round(1)
+    return average
+  end
```

Comment:
You can clean this up with enumerable methods ([`compact`](https://ruby-doc.org/core-2.6.5/Array.html#method-i-compact) is actually an array method):

```suggestion
  def average_rating
    ratings = self.trips.compact.map { |t| t.rating.to_f }
    
    return (ratings.sum / ratings.length).round(1)
  end
```